app/views/shared/reports/_grouped_table.html.haml

```diff
@@ -2,7 +2,7 @@
   = render partial: "shared/reports/table_empty", locals: { type: type }
 - else
   %table.govuk-table
-    = render partial: "shared/reports/table_header", locals: { type: type }
+    = render partial: "shared/reports/table_header", locals: { type: type, show_organisation: current_user.service_owner? }
```

Comment:
I think it'd be good to test this in feature specs, as they're often the only way of testing view logic. Would that be a pain to add here?

Reply:
I've added an additional commit to cover the headings. I toyed with restructuring the whole file into different routes, but it would've messed with some of the existing contexts, and some tests cover multiple routes

Have a couple of concerns:
- I wonder if it's a little complex, although it was already a somewhat abstracted spec
- I've used `HideFromBullet` to avoid an eager load error, but previously we haven't used this in specs. Not sure if it's worth adding it to the controller itself

Would you mind taking a second look?